posydon/popsyn/synthetic_population.py

```diff
@@ -224,8 +224,84 @@ def merge_parallel_runs(self, pop):
 # Helper classes #
 ##################
 
+class DFInterface:
+    """A class to handle the interface between the population file and the History and Oneline classes."""
+    
+    def __init__():
+        pass
+
+    def head(self, key, n=10):
+        """Return the first n rows of the key table
+
+        Parameters
+        ----------
+        key: str
+            The key of the table
+        n : int, optional
+            The number of rows to return. Default is 10.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The first n rows of the key table.
+
+        """
+        with pd.HDFStore(self.filename, mode="r") as store:
+            return store.select(key, start=0, stop=n)
+        
+    def tail(self, key, n=10):
+        """
+        Get the last n rows of the key table.
+
+        Parameters
+        ----------
+        key : str
+            The key of the table.
+        n : int, optional
+            The number of rows to return. Default is 10.
 
-class History:
+        Returns
+        -------
+        pd.DataFrame
+            The last n rows of the key table.
+        """
+        with pd.HDFStore(self.filename, mode="r") as store:
+            return store.select(key, start=-n)
+        
+    
+    def select(self, key, where=None, start=None, stop=None, columns=None):
+        '''Select a subset of the key table based on the given conditions.
+        
+        Parameters
+        ----------
+        key : str
+            The key of the table to select from.
+        where : str, optional
+            A string representing the query condition to apply to the data.
+        start : int, optional
+            The starting index of the data to select.
+        stop : int, optional
+            The ending index of the data to select.
+        columns : list, optional
+            A list of column names to select.
+        
+        Returns
+        -------
+        pandas.DataFrame
+            The selected data as a DataFrame.
+        '''
+        # we have to chunk the read because of memory issues
+        with pd.HDFStore(self.filename, mode="r") as store:
+            iterator = store.select("history", where=where, start=start, stop=stop, columns=columns, chunksize=self.chunksize)
```

Comment:
Shouldn't we have `key` as the first argument instead of "history"?

Reply:
changed